<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>DialogManageRoutes</class>
 <widget class="QDialog" name="DialogManageRoutes">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Routes</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Common</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QWidget" name="widget" native="true">
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="1" column="1">
           <widget class="QComboBox" name="sniffing_mode">
            <item>
             <property name="text">
              <string>Disable</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Sniff result for routing</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>Sniff result for destination</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label">
            <property name="text">
             <string>Sniffing Mode</string>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Server Address Strategy</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QComboBox" name="domainStrategyCombo">
            <property name="editable">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="4" column="0">
           <widget class="QLabel" name="route_prof_l">
            <property name="text">
             <string>Routing Profile</string>
            </property>
           </widget>
          </item>
          <item row="4" column="1">
           <widget class="QComboBox" name="route_prof"/>
          </item>
          <item row="3" column="1">
           <widget class="QComboBox" name="outbound_domain_strategy">
            <property name="editable">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_6">
            <property name="toolTip">
             <string notr="true">For V2Ray, it sets routing.domainStrategy
For sing-box, it sets inbound.domain_strategy</string>
            </property>
            <property name="text">
             <string>Domain Strategy</string>
            </property>
           </widget>
          </item>
          <item row="5" column="1">
           <widget class="QComboBox" name="default_out">
            <item>
             <property name="text">
              <string>proxy</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>direct</string>
             </property>
            </item>
           </widget>
          </item>
          <item row="5" column="0">
           <widget class="QLabel" name="default_out_l">
            <property name="text">
             <string>Default Outbound</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>DNS</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QGroupBox" name="simple_dns_box">
         <property name="title">
          <string>Simple DNS Settings</string>
         </property>
         <layout class="QGridLayout" name="gridLayout">
          <item row="1" column="0">
           <widget class="QLabel" name="label_8">
            <property name="toolTip">
             <string>This is especially important and it is recommended to use the default value of &quot;localhost&quot;.
If the default value does not work, try changing it to &quot;223.5.5.5&quot;.
For more information, see the document &quot;Configuration/DNS&quot;.</string>
            </property>
            <property name="text">
             <string>Direct DNS</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <layout class="QHBoxLayout" name="horizontalLayout_7">
            <item>
             <widget class="QComboBox" name="remote_dns">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="editable">
               <bool>true</bool>
              </property>
              <item>
               <property name="text">
                <string>8.8.8.8</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">https://8.8.8.8/dns-query</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">https://1.0.0.1/dns-query</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">https://1.1.1.1/dns-query</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">https://dns.google/dns-query</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">https://one.one.one.one/dns-query</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">https://[2001:4860:4860::8888]/dns-query</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">https://[2606:4700:4700::1111]/dns-query</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Query Strategy</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="remote_dns_strategy"/>
            </item>
           </layout>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="label_9">
            <property name="text">
             <string>Remote DNS</string>
            </property>
            <property name="buddy">
             <cstring>widget</cstring>
            </property>
           </widget>
          </item>
          <item row="2" column="2">
           <layout class="QHBoxLayout" name="horizontalLayout_3">
            <item>
             <widget class="QCheckBox" name="dns_routing">
              <property name="text">
               <string>Enable DNS Routing</string>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="horizontalSpacer_2">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>40</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="dns_final_out_l">
              <property name="text">
               <string>Final DNS Out</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="dns_final_out">
              <item>
               <property name="text">
                <string notr="true">proxy</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">direct</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="2">
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QComboBox" name="direct_dns">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="editable">
               <bool>true</bool>
              </property>
              <item>
               <property name="text">
                <string notr="true">localhost</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">https://223.5.5.5/dns-query</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">https://1.12.12.12/dns-query</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">https://dns.alidns.com/dns-query</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">https://doh.pub/dns-query</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">223.5.5.5</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">119.29.29.29</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">2400:3200::1</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string notr="true">2402:4e00::</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Query Strategy</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="direct_dns_strategy"/>
            </item>
           </layout>
          </item>
         </layout>
        </widget>
       </item>
       <item>
        <widget class="QGroupBox" name="groupBox_2">
         <property name="title">
          <string>DNS Object Settings</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <widget class="QCheckBox" name="use_dns_object">
              <property name="text">
               <string>Use DNS Object</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="format_dns_object">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Format</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="dns_document">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>Document</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPlainTextEdit" name="dns_object"/>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Route</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QGroupBox" name="route_profiles_box">
         <property name="title">
          <string>Routing Profiles</string>
         </property>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QListWidget" name="route_profiles"/>
          </item>
          <item>
           <widget class="QWidget" name="widget_2" native="true">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QPushButton" name="new_route">
               <property name="text">
                <string>New</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="edit_route">
               <property name="text">
                <string>Edit</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="delete_route">
               <property name="text">
                <string>Delete</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>tabWidget</tabstop>
  <tabstop>sniffing_mode</tabstop>
  <tabstop>domainStrategyCombo</tabstop>
  <tabstop>outbound_domain_strategy</tabstop>
  <tabstop>remote_dns</tabstop>
  <tabstop>remote_dns_strategy</tabstop>
  <tabstop>direct_dns</tabstop>
  <tabstop>direct_dns_strategy</tabstop>
  <tabstop>dns_routing</tabstop>
  <tabstop>dns_final_out</tabstop>
  <tabstop>use_dns_object</tabstop>
  <tabstop>format_dns_object</tabstop>
  <tabstop>dns_document</tabstop>
  <tabstop>dns_object</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>DialogManageRoutes</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>DialogManageRoutes</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>399</x>
     <y>574</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
